newsfragments/1850.bugfix.rst

```diff
@@ -0,0 +1 @@
+Update codebase to more consistently use the ``eth_typing.BlockNumber`` type.
```

Comment:
*Nitpick*, this might be better served as `misc` as it's more of a refactoring than a bugfix. At least from the user perspective it's not fixing anything worth showing up in the release notes. Now that we use towncrier we get much longer [release notes](https://trinity-client.readthedocs.io/en/latest/release_notes.html#trinity-0-1-0-alpha-28-2019-09-12) than before which is great but we should also try to keep them as meaningful and focused as we can to not overwhelm readers with less relevant information.

Reply:
I marked it as bugfix because in other libraries if you do `from eth.constants import GENESIS_BLOCK_NUMBER` you end up having to do `BlockNumber(GENESIS_BLOCK_NUMBER)` to have it work appropriately with properly typed interfaces.  This fixes that.